generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -168,6 +168,14 @@ private Expr makeOpaqueZeroOrOne(boolean isZero, BasicType type, boolean constCo
           }
           return new FunctionCallExpr("sqrt", makeOpaqueZeroOrOne(isZero, type, constContext,
                 newDepth, fuzzer));
+        case 3:
+          // represent 1 as the length of normalized vector
+          if (!BasicType.allGenTypes().contains(type) || isZero) {
+            continue; // normalize doesn't operate on non-gen types.
```

Comment:
Given that the comment is explaining the case for "continue", it would be good to expand this to say explicitly that the isZero case cannot be satisfied by this option.

However, I would actually slightly prefer:

if (isZero) {
  continue; // length(normalize(...)) only provides a means of representing 1, not 0
}

if (!Basictype.allGenTypes().contains(type)) {
  continue; // ...
}

That might seem more laboured, but I think it more cleanly separates the reasons for continuing.

Reply:
Done.